src/Graphics/Shader/Fragment.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE FlexibleInstances #-}
+module Graphics.Shader.Fragment where
+
+import Control.Applicative (liftA2)
+import Control.Applicative.Free.Freer
+import Linear.V2
+import Linear.V4
+
+type Colour = V4
+
+data FragmentF f
+  = Coord (V4 Float -> f)
+  | SampleID (Int -> f)
+  | NumSamples (Int -> f)
+  | PointCoord (V2 Float -> f)
+  | SamplePosition (V2 Float -> f)
+  | SetDepth Float f
+  | SetColour (Colour Float) f
```

Comment:
This is almost certainly wrong. Specifically, we want to compute program values here.

I don’t think doing this in a free applicative functor is all that helpful at the moment, either; this should be defined as a (directly recursive) GADT instead.